plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/FileSystemExchangeManager.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.exchange;
+
+import io.trino.spi.TrinoException;
+import io.trino.spi.exchange.Exchange;
+import io.trino.spi.exchange.ExchangeContext;
+import io.trino.spi.exchange.ExchangeManager;
+import io.trino.spi.exchange.ExchangeSink;
+import io.trino.spi.exchange.ExchangeSinkInstanceHandle;
+import io.trino.spi.exchange.ExchangeSource;
+import io.trino.spi.exchange.ExchangeSourceHandle;
+
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.inject.Inject;
+
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.security.NoSuchAlgorithmException;
+import java.util.List;
+import java.util.Optional;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static io.trino.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+
+public class FileSystemExchangeManager
+        implements ExchangeManager
+{
+    private final ExchangeStorage exchangeStorage;
+    private final URI baseDirectory;
+    private final boolean exchangeEncryptionEnabled;
+
+    @Inject
+    public FileSystemExchangeManager(ExchangeStorage exchangeStorage, ExchangeConfig exchangeConfig)
+    {
+        this.exchangeStorage = exchangeStorage;
+        try {
+            this.baseDirectory = new URI(exchangeConfig.getBaseDirectory() + "/");
+        }
+        catch (URISyntaxException e) {
+            throw new IllegalArgumentException(e);
+        }
+        this.exchangeEncryptionEnabled = exchangeConfig.isExchangeEncryptionEnabled();
+    }
+
+    @Override
+    public Exchange create(ExchangeContext context, int outputPartitionCount)
+    {
+        Optional<SecretKey> secretKey = Optional.empty();
+        if (exchangeEncryptionEnabled) {
+            try {
+                KeyGenerator keyGenerator = KeyGenerator.getInstance("AES");
+                keyGenerator.init(256);
+                secretKey = Optional.of(keyGenerator.generateKey());
+            }
+            catch (NoSuchAlgorithmException e) {
+                throw new TrinoException(GENERIC_INTERNAL_ERROR, "Failed to generate new secret key: " + e.getMessage(), e);
+            }
+        }
+        FileSystemExchange exchange = new FileSystemExchange(baseDirectory, exchangeStorage, context, outputPartitionCount, secretKey);
+        exchange.initialize();
```

Comment:
Any reason to not just do initialization in the constructor?

Reply:
I feel that class constructors in Trino typically only do class member assignments? Of course, there is nothing preventing us from doing the initialization in constructor